cli/tests/sample-configs/valid/scope_and_when.toml

```diff
@@ -0,0 +1,10 @@
+#:schema ../../../src/config-schema.json
+
+# --when can be used at the top-level. Useful when config is organized over
+# several files.
+--when.commands = ["log"]
+user.name = "John Doe"
+
+[[--scope]]
+--when.repositories = ["~/repos/employer"]
```

Comment:
(Optional) Perhaps switch the nesting? IIUC, this will trigger only for the "log" command in employer repos. It'd make more sense to change the email for all commands in employer repos.

Also, perhaps add a comment that this is the semantics?

If I didn't understand the semantics, a comment would be even more useful.